lib/src/summarize.dart

```diff
@@ -4,63 +4,302 @@
+      'grouping of letters'
+    ],
+    'errorCount-2': [
+      'many',
+      'a motherload of',
+      'copious amounts of',
+      'unholy quantities of'
+    ],
+    'errorCount-1': [
+      'some',
+      'a few',
+      'sparse amounts of',
+      'very few instances of'
+    ],
+    'errorCount-0': ['zero', 'no', 'a nonexistent amount of', '0.0000', '0'],
+    'use': ['demonstrates', 'contains', 'illustrates', 'depicts'],
+    'code-0': ['it'],
+    'code-1': ['It'],
+  };
+
+  Summarizer(
+      {this.dart, this.html, this.css, this.analysis}) {
+    if (dart == null) throw new ArgumentError('Input cannot be null.');
     resultsPresent = !(analysis == null);
-    storage = new _SummarizeToken(input, analysis);
+    MD5 encryptor = new MD5();
+    encryptor.add(dart.codeUnits);
+    randomizer = _sumList(encryptor.close());
+    storage = new _SummarizeToken(dart, analysis : analysis);
+  }
+
+  int _sumList(List<int> list) {
+    int sum = 0;
+    for (int number in list) {
+      sum += number;
+    }
+    return sum;
+  }
+
+  String _categorySelector(String category, int itemCount) {
+    if (category == 'size' || category == 'errorCount') {
+      List<int> maxField = cuttoffs[category];
+      int counter = 0;
+      while (counter < maxField.length) {
+        if (itemCount < maxField[counter]) {
+          return '${category}-${counter}';
+        }
+        counter++;
+      }
+      return '${category}-${counter - 1}';
+    } else {
+      return null;
+    }
+  }
+
+  String _wordSelector(String category) {
+    if (categories.containsKey(category)) {
+      List<String> returnSet = categories[category];
+      return returnSet.elementAt(randomizer % returnSet.length);
+    } else {
+      return null;
+    }
+  }
+
+  String _sentenceFiller(String word, [int size]) {
+    if (size != null) {
+      return _wordSelector(_categorySelector(word, size));
+    } else {
+      return _wordSelector(word);
+    }
+  }
+
+  String _additionList(List<String> list) {
+    if (list.length == 0) return '';
+    String englishList = ' Also, mentions ';
+    for (int i = 0; i < list.length; i++) {
+      englishList += list[i];
+      if (i < list.length - 2) {
+        englishList += ', ';
+      }
+      if (i == list.length - 2) {
+        if (i != 0) {
+          englishList += ',';
+        }
+        englishList += ' and ';
+      }
+    }
+    englishList += '. ';
+    return englishList;
+  }
+  
+  ///Testing add-ons
+  List<String> additionSearch() => _additionSearch();
+  
+  List<String> _additionSearch() {
+    List<String> features = new List<String>();
+    for (String feature in additionKeyWords.keys) {
+      if (dart.contains(feature)) {
+        features.add(additionKeyWords[feature]);
+      }
+    }
+    return features;
+  }
+
+  List<String> _codeSearch() {
+    List<String> features = new List<String>();
+    for (String feature in codeKeyWords.keys) {
+      if (dart.contains(feature)) {
```

Comment:
Perhaps `dart.contains()` can be a separate method? It's used twice, and it might get more complicated in the future. Something like `bool _usedInDartSource(String feature)`